File: tests/fixtures/edgecase_repo/rust/src/module7.rs
use crate::module6::Service6;

pub struct Base7 {
    pub name: String,
}

pub struct Service7 {
    pub base: Base7,
}

pub trait Handler7 {
    fn handle(&self, payload: &str) -> bool;
}

impl Handler7 for Service7 {
    fn handle(&self, payload: &str) -> bool {
        self.validate(payload);
        true
    }
}

impl Service7 {
    fn validate(&self, payload: &str) {}
}

pub fn helper_7(value: &str) -> &str {
    value
}

File: tests/fixtures/edgecase_repo/javascript/src/module7.js
const { Service6 } = require("./module6");

class Base7 {
  constructor(name) {
    this.name = name;
  }
}

class Service7 extends Base7 {
  handle(payload) {
    this.validate(payload);
    return new Service6("x").handle(payload);
  }

  validate(payload) {
    return payload !== null;
  }
}

function helper7(value) {
  return value;
}

module.exports = { Service7 };
